File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/storage/ingrediente/IngredienteStorageServiceCSV.java
package storage.ingrediente;

import config.ConfigApp;
import models.Ingrediente;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class IngredienteStorageServiceCSV implements IngredienteStorageService{

    ConfigApp configApp = ConfigApp.getInstance();

    File file = new File(configApp.getAPP_DATA()+File.separator+"ingrediente.csv");

    public IngredienteStorageServiceCSV() throws IOException {
    }

    @Override
    public void saveAll(List<Ingrediente> entities) {
        try(BufferedWriter br = new BufferedWriter(new FileWriter(file.getPath()))) {
            br.write("id,nombre,precio\n");
            entities.forEach(ingrediente ->
                    {
                        try {
                            br.append(
                                    ingrediente.getId()+","+ingrediente.getNombre()+","+ ingrediente.getPrecio()+"\n"
                            );
                        } catch (IOException e) {
                            throw new RuntimeException(e);
                        }
                    }
            );
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public List<Ingrediente> loadAll() {
        List<Ingrediente> ingredientes = new ArrayList<>();
        if(!file.exists()) return ingredientes;
        try(BufferedReader br = new BufferedReader(new FileReader(file))){
            String line = br.readLine();
            while ((line = br.readLine()) != null) {
                String[] campos = line.split(",");
                ingredientes.add(new Ingrediente(Integer.parseInt(campos[0]), campos[1], Double.parseDouble(campos[2])));
            }
        } catch (FileNotFoundException e) {
            throw new RuntimeException(e);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        return ingredientes;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/storageService/Ingrediente/IngredienteStorageServiceCSV.kt
package storageService.Ingrediente

import config.ConfigApp
import model.Ingrediente
import mu.KotlinLogging
import java.io.File

class IngredienteStorageServiceCSV: IngredienteStorageService {

    private val config = ConfigApp
    private val file = File(config.APP_DATA+File.separator+"ingredientes.csv")
    private val logger = KotlinLogging.logger {}

    override fun saveAll(entities: List<Ingrediente>) {
        logger.debug { "Se guardan todos los ingredientes en el fichero CSV." }
        file.writeText("id,nombre,precio"+"\n")
        entities.forEach { ingrediente ->
            file.appendText("${ingrediente.id},${ingrediente.nombre},${ingrediente.precio}"+"\n")
        }
    }

    override fun loadAll(): List<Ingrediente> {
        logger.debug { "Se cargán todos los ingredientes del fichero CSV." }
        if(!file.exists()) return emptyList()
        return file.readLines().drop(1).map { linea -> linea.split(",") }
            .map { campo -> campo.map { it.trim() } }
            .map { campo ->
                Ingrediente(
                    id = campo[0].toInt(),
                    nombre = campo[1],
                    precio = campo[2].toDouble()
                )
            }
    }
}